operations/agent_monitor.go

```diff
@@ -0,0 +1,427 @@
+			},
+			cli.StringFlag{
+				Name:  clientPathFlagName,
+				Usage: "the name of the agent's evergreen binary",
+			},
+			cli.StringFlag{
+				Name:  logPrefixFlagName,
+				Value: monitorLoggerName,
+				Usage: "the prefix for the monitor's log name",
+			},
+			cli.IntFlag{
+				Name:  jasperPortFlagName,
+				Value: defaultJasperPort,
+				Usage: "the port that is running the Jasper RPC service",
+			},
+			cli.IntFlag{
+				Name:  portFlagName,
+				Value: defaultAgentStatusPort - 1,
+				Usage: "the port that used by the monitor",
+			},
+		},
+		Before: mergeBeforeFuncs(
+			requireStringFlag(clientURLFlagName),
+			requireStringFlag(clientPathFlagName),
+			requireIntValueBetween(jasperPortFlagName, 0, 1<<16),
+			requireIntValueBetween(portFlagName, 0, 1<<16),
+			func(*cli.Context) error {
+				grip.SetName(monitorLoggerName)
+				return nil
+			},
+		),
+		Action: func(c *cli.Context) error {
+			m := &monitor{
+				certificatePath: c.String(certificatePathFlagName),
+				clientURL:       c.String(clientURLFlagName),
+				clientPath:      c.String(clientPathFlagName),
+				jasperPort:      c.Int(jasperPortFlagName),
+				port:            c.Int(portFlagName),
+				logPrefix:       c.String(logPrefixFlagName),
+			}
+
+			args, err := getAgentArgs(c, os.Args)
+			if err != nil {
+				return errors.Wrap(err, "error getting agent args")
+			}
+			m.agentArgs = args
+
+			if err = setupLogging(m); err != nil {
+				return errors.Wrap(err, "error setting up logging")
+			}
+
+			// Reserve the given port to prevent other monitors from starting.
+			if _, err = net.Listen("tcp", fmt.Sprintf(":%d", m.port)); err != nil {
+				if err == syscall.EADDRINUSE {
+					return nil
+				}
+				return errors.Wrapf(err, "failed to listen on port %d", m.port)
+			}
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			go handleMonitorSignals(ctx, cancel)
+
+			client, err := setupJasperConnection(ctx, m, defaultRetryArgs())
+			if err != nil {
+				return errors.Wrap(err, "failed to connect to RPC service")
+			}
+			defer func() {
+				grip.Error(errors.Wrap(client.CloseConnection(), "failed to close RPC client connection"))
+			}()
+			m.jasperClient = client
+
+			runMonitor(ctx, m)
+
+			return nil
+		},
+	}
+}
+
+// getAgentArgs find the slice of flags that should be forwarded to the agent.
+func getAgentArgs(c *cli.Context, args []string) ([]string, error) {
+	beginIndex, endIndex := -1, -1
+
+	for index, arg := range args {
+		if arg == "agent" {
+			beginIndex = index
+		}
+		if arg == c.Command.Name {
+			endIndex = index
+		}
+	}
+
+	if beginIndex == -1 || endIndex == -1 {
+		return nil, errors.Errorf("agent args are invalid: %v", args)
+	}
+
+	return args[beginIndex+1 : endIndex], nil
+}
+
+func retryWithArgs(ctx context.Context, retry util.RetriableFunc, args *retryArgs) error {
+	return util.Retry(ctx, retry, args.maxAttempts, args.minDelay, args.maxDelay)
```

Comment:
this should probably be in the utils package. I'm not super fond of our retryable functions, but these kind of one off wrappers have a way of proliferating... either get rid of this and retryArgs entirely or move it into util.

also I wouldn't pass a struct like retry args as a pointer...